src/Bicep.Cli/Program.cs

```diff
@@ -129,7 +130,7 @@ private bool LogDiagnosticsAndCheckSuccess(IDiagnosticLogger logger, Compilation
 
         private void BuildSingleFile(IDiagnosticLogger logger, string bicepPath, string outputPath)
         {
-            var syntaxTreeGrouping = SyntaxTreeGroupingBuilder.Build(new FileResolver(), bicepPath);
+            var syntaxTreeGrouping = SyntaxTreeGroupingBuilder.Build(new FileResolver(), new Workspace(), bicepPath);
```

Comment:
This contract seems inverted somehow. In Roslyn, I believe you construct the workspace first and then add things to it. The workspace there is immutable so the Add() methods just return a new workspace with the change made. (The part that is the weirdest to me is the syntax tree grouping builder being on the outside of the expression.)

Any chance we can make it more like the Roslyn contract? Although their workspace does appear to be mutable in this example: https://riptutorial.com/roslyn/example/30067/creating-an-adhocworkspace-and-adding-a-new-project-and-a-file-to-it-.

Reply:
I want to gravitate in the direction of the Roslyn Workspace, but didn't want to make all of the changes in this PR and have it turn into a giant refactor. At the moment the Workspace just encompasses a collection of `SyntaxTree` nodes (to avoid duplication). In the future I think we could consider moving SyntaxTreeGroupings (at some point, I should give this class a better name), SemanticModels, and Compilations into the Workspace. I didn't see a huge benefit of making it immutable just yet, but feel like it might be a worthwhile change to make when we are doing more concurrent processing.

The reason you're seeing the SyntaxTreeGrouping on the outside of the Workspace is just because of the above.

Reply:
Makes sense. I mainly want our final-ish contract to be super clean. Definitely ok if it takes a few PRs to get there.